test/shopify.spec.js

```diff
@@ -272,4 +277,318 @@ describe("shopify format", () => {
       ).toBe("This is my 2st car");
     });
   });
+
+  describe("with react-i18next (t)", () => {
```

Comment:
This is a good first PR to introduce tests for `react-i18next`, but there's more functionality to the `<Trans />` component that will need to be added to [use it with React components](https://react.i18next.com/latest/trans-component#using-with-react-components):
* [Alternative usage which lists the components](https://react.i18next.com/latest/trans-component#alternative-usage-which-lists-the-components-v11.6.0)
  * We might want to consider only supporting this option, because it's simpler, and it discourages leaving "default" hard-coded text in the `.tsx` files.
* [Rules for default usage](https://react.i18next.com/latest/trans-component#how-to-get-the-correct-translation-string)
  * Needs to also take into consideration the [settings for simple HTML elements](https://react.i18next.com/latest/trans-component#usage-with-simple-html-elements-like-less-than-br-greater-than-and-others-v10.4.0) and some other interpolation rules.

Maybe it's possible to copy-paste a lot of this from `react-i18next` itself, so there won't be a lot to it 🤞🏻